### PR TITLE
Kubernetes plugin:  Loading default Kube configuration 

### PR DESCRIPTION
**Pull request by kim5566**
## Hey, I just made a Pull Request!

Load default kubeconfig from cluster if KSA token not provided.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/kubernetes-backend/src/service/KubernetesClientProvider.ts
Comment (freben) [inline plugins/kubernetes-backend/src/service/KubernetesClientProvider.ts:48]: Hm. Does this need to be updated now, after https://github.com/backstage/backstage/pull/9046 so that it checks for `googleServiceAccount` as well? Or am I mixing stuff up

Comment (kim5566) [reply plugins/kubernetes-backend/src/service/KubernetesClientProvider.ts:48]: hi! This one will use RBAC auth.

and  googleServiceAccount will use GCP roles (GCP IAM).

So still needed?

Comment (freben) [reply plugins/kubernetes-backend/src/service/KubernetesClientProvider.ts:48]: I am not sure, I think I'll let you decide. Should `loadFromOptions` or `loadFromDefault` be called on GCP? I'm happy to merge after just settling that last question :)

Comment (kim5566) [reply plugins/kubernetes-backend/src/service/KubernetesClientProvider.ts:48]: :) sure. 

`loadFromOptions` or `loadFromDefault` will be called on GKE. Line 48 was using the **Kubernetes service account** token.  The permission is defined by ClusterRole and RoleBindings.

#9046 add the auth provider use **Google Service account.**  token. The access is controlled by the GCP IAM roles using workload identities on cluster not RBAC. 

So basically, they both have same name called service account but KSA and GSA are looking for different credentials.

**Review by freben — APPROVED** [review]

**Pull request merged**
